File: src/main/java/dev/retrotv/crypto/twe/des/TripleDESCBC.java
package dev.retrotv.crypto.twe.des;

import java.security.Key;
import java.security.NoSuchAlgorithmException;

import javax.crypto.KeyGenerator;
import javax.crypto.spec.IvParameterSpec;

import dev.retrotv.crypto.exception.KeyGenerateException;
import dev.retrotv.crypto.twe.ParameterSpecGenerator;
import dev.retrotv.utils.SecureRandomUtil;

import static dev.retrotv.enums.CipherAlgorithm.TRIPLE_DESCBC;

public class TripleDESCBC extends DES implements ParameterSpecGenerator<IvParameterSpec> {
    public TripleDESCBC() {
        this.algorithm = TRIPLE_DESCBC;
    }

    @Override
    public Key generateKey() throws KeyGenerateException {
        try {
            KeyGenerator keyGenerator = KeyGenerator.getInstance("DESede");
            return keyGenerator.generateKey();
        } catch (NoSuchAlgorithmException e) {
            throw new KeyGenerateException("NoSuchAlgorithmException: \n지원하지 않는 암호화 알고리즘 입니다.");
        }
    }

    @Override
    public IvParameterSpec generateSpec() {
        return new IvParameterSpec(SecureRandomUtil.generate(8));
    }
}

File: src/main/kotlin/dev/retrotv/crypto/twe/des/TripleDESCBC.kt
package dev.retrotv.crypto.twe.des

import dev.retrotv.crypto.twe.ParameterSpecGenerator
import dev.retrotv.enums.Algorithm
import dev.retrotv.utils.generate
import javax.crypto.spec.IvParameterSpec

/**
 * TripleDES/CBC 양방향 암호화 클래스 입니다.
 *
 * @author  yjj8353
 * @since   1.0.0
 */
class TripleDESCBC : TripleDES(), ParameterSpecGenerator<IvParameterSpec> {

    init {
        algorithm = Algorithm.Cipher.TRIPLE_DESCBC
    }

    override fun generateSpec(): IvParameterSpec {
        return IvParameterSpec(generate(8))
    }
}
